ui/src/criteria/attribute.tsx

```diff
@@ -17,6 +23,12 @@ type Selection = {
   name: string;
 };
 
+type DataRange = {
+  id: string;
+  min: number | undefined;
```

Comment:
min/max previously allowed undefined because that was the case for non-range selection. There's no need to allow it now since an undefined dataRange indicates that. That will clean up places where you have to test min/max for undefined and can instead test dataRanges.